### ISSUE TITLE
Questions about the DeepSort train method

### ISSUE THREAD

**Issue by youngjae-avikus** [mikel-brostrom/Yolov5_StrongSORT_OSNet#631]
### Search before asking

- [X] I have searched the Yolov5_StrongSORT_OSNet [issues](https://github.com/mikel-brostrom/Yolov5_StrongSORT_OSNet/issues) and found no similar bug report.

### Question

Hello, I am referring to the deepsort currently implemented in 5.0 tag

https://github.com/mikel-brostrom/Yolov5_StrongSORT_OSNet/tree/v5.0

In the official code, he focused on generative learning using `cosine_softmax loss`

![image](https://user-images.githubusercontent.com/99321359/205919051-f176b4a5-bfe5-4eb3-9dfb-9e98a5c49356.png)

However, in the code implemented by pytorch that code refer to, it simply conducted a discrimitive learning with cross_entropy

https://github.com/mikel-brostrom/Yolov5_StrongSORT_OSNet/blob/790f02f78ed54fef97d3c58d42a2d601cda12652/deep_sort_pytorch/deep_sort/deep/train.py#L72

1. Is there not much difference in the performance of the results for the two learning methods introduced?
2. When calculating cost_matrix, you calculated it as Euclidean distance, not cosine similarity, but is this closely related to cross_entropy learning?

Thank U

**Comment by youngjae-avikus** [comment]
> > Is there not much difference in the performance of the results for the two learning methods introduced?
> 
> This is a question for this repo https://github.com/KaiyangZhou/deep-person-reid
> 
> > When calculating cost_matrix, you calculated it as Euclidean distance, not cosine similarity, but is this closely related to cross_entropy learning?
> 
> [KaiyangZhou/deep-person-reid#502](https://github.com/KaiyangZhou/deep-person-reid/issues/502)

Yes, I understand. In short,
1. regardless of Euclidean,Cosine distance loss, it seems right to train with cross-entroy loss
2. There was an update time difference, and in the end, cosine similarity produced good performance regardless of domain, right?

**Comment by mikel-brostrom** [comment]
1. It is a classification problem so yes
2. the dot product between image features and label vectors (i.e. the classification weights) essensially measures their similarity in the cosine space, therefore, using cosine distance makes more sense and often performs better
